### PR TITLE
Layer management syst.... I mean layer

### PR DESCRIPTION
**Pull request by Burke-Daniel**
Hot swappable layers. That's it.

You can either use the pg up/pg down keys, or f/b to move forwards and backwards through layers, it will also cycle so if you try to go "up" on the last layer in the list, it will bring you back to the first layer.

Note there are pong layer changes in here that I made, but then cherry picked so they should also be on the pong branch.

### CODE REVIEW

File: Sandbox/ManagerLayer.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	}
+
+	if (!m_ManagedLayers.empty())
+	{
+		m_ManagedLayers.front()->activate();
+		m_ActiveLayerIndex = 0;
+	}
+}
+
+void ManagerLayer::onEvent(const Aquarius::Event& event)
```
Comment (liamjwarren) [inline Sandbox/ManagerLayer.cpp:30]: We need the esc to exit out of the application, with the mouse disabled it makes it difficult to navigate and close the window.

**Review by liamjwarren — CHANGES_REQUESTED** [review]
Looks good, just the esc key for easy exiting of the application.

File: Sandbox/ManagerLayer.cpp (hunk 1 above)
Comment (colton-smith) [reply Sandbox/ManagerLayer.cpp:30]: @Burke-Daniel Esc key callback in window should handle this right?

File: Sandbox/ManagerLayer.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+				m_ManagedLayers[++m_ActiveLayerIndex]->activate();
+			}
+			else
+			{
+				m_ManagedLayers[m_ActiveLayerIndex]->deactivate();
+				m_ActiveLayerIndex = 0;
+				m_ManagedLayers[m_ActiveLayerIndex]->activate();
+			}
+			break;
+		}
```
Comment (colton-smith) [inline Sandbox/ManagerLayer.cpp]: indentation is a little off here

File: Sandbox/PongLayer.cpp [hunk 3]
```diff
@@ -135,8 +141,8 @@ void PongLayer::onUpdate(Aquarius::timeDelta_t dt)
 	m_RightPaddle.Render();
 	m_Ball.Render();
 
-	AQ_INFO("Frametime: %v ms", dt);
-	AQ_INFO("FPS: %v", 1000 / dt);
+	AQ_TRACE("Frametime: %v ms", dt);
+	AQ_TRACE("FPS: %v", 1000 / dt);
```
Comment (colton-smith) [inline Sandbox/PongLayer.cpp]: Where do we want to set the log level for the demo application?

File: Sandbox/Sandbox.cpp [hunk 4]
```diff
@@ -91,7 +93,14 @@ void SandboxLayer::onUpdate(Aquarius::timeDelta_t ts)
...
 {
-    PushLayer(std::make_unique<PongLayer>());
+    Aquarius::Log::setLogLevel(Aquarius::LogLevel::Info);
+
+    auto testLayer1 = PushLayer(std::make_unique<SandboxLayer>());
+    auto testLayer2 = PushLayer(std::make_unique<PongLayer>());
+
+    PushLayer(std::make_unique<ManagerLayer>(
+        std::vector<Aquarius::Layer*>{ testLayer1, testLayer2 }));
+
```
Comment (colton-smith) [inline Sandbox/Sandbox.cpp:103]: Nice!

**Review by colton-smith — APPROVED** [review]
Literally just indentation in the switch case of ManagerLayer::OnEvent is warped, then GTM

File: Sandbox/ManagerLayer.cpp (hunk 1 above)
Comment (liamjwarren) [reply Sandbox/ManagerLayer.cpp:30]: Yep, after reviewing this PR I went and did the pong one and made a note to put the esc back in the window haha

**Review by liamjwarren — APPROVED** [review]

**Pull request merged**